### DISCUSSION TITLE
How to add extra button inside mv-popup?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#906] [Q&A] [answered]
Hi

I'm trying to add an extra button while editing images, inside mv-popup

<img width="451" alt="image" src="https://user-images.githubusercontent.com/89095737/186146592-79520c01-be38-47b0-b12a-a4ef364a290a.png">

I guess best approach would be to use Mavo.Elements.register() , as cropper plugin is using, but

unfortunately the documentation for adding a new element type is not ready yet,

so in my[ codepen here](https://codepen.io/galinhalx/pen/KKoEOYL)

I tried using insertAdjacentHTML in javascript, looking for mv-upload-popup, 

or even using mv-action directly on <img> (as on page load, popup is not present yet)

...but it doesn't want to work...

Am I missing something? Or is this not possible the way I'm trying to do?

My goal is to add an extra button, to let user be able to search for a photo in unsplash service

**Comment by DmitrySharabin** [comment, marked as answer]
Hey there,

I would suggest using the `primitive-createuploadpopup-beforereturn` Mavo hook (the one the Cropper plugin is based on).

You might find helpful [this pen](https://codepen.io/dmitrysharabin/pen/ExEJKRv) and the [Bliss docs](https://blissfuljs.com/docs.html#fn-create).

**Reply by GalinhaLX** [reply]
Thanks Dmitry!
